tests/test_searcher.py

```diff
@@ -430,22 +437,57 @@ def test_ordered_moves_end_game(
 
         for i, move in enumerate(all_moves):
             mo = MvvLvaHeuristic()
-            score = mo.evaluate(board, move)
+            score = mo.evaluate(board, move, 0)
             assert score == move_scores[i]
 
     def test_sorting_legal_moves(
         self, _init_searcher: Searcher, fen_string: str, move_scores: list[int]
     ) -> None:
-        """
-        Test sorting of legal moves by mvv_lva_heuristic
-        """
         board = init_board(fen_string)
         s = _init_searcher
 
-        legal_moves = s._ordered_moves(board, board.legal_moves)
+        legal_moves = s._ordered_moves(board, board.legal_moves, 0)
 
         num_moves = len(move_scores)
         for i, move in enumerate(legal_moves):
             mo = MvvLvaHeuristic()
-            score = mo.evaluate(board, move)
+            score = mo.evaluate(board, move, 0)
             assert score == move_scores[num_moves - i - 1]
+
+
+@pytest.mark.parametrize(
+    ("fen_string", "move_scores"),
+    [
+        ("k7/3q4/8/K7/2N5/8/8/8 w - - 1 34", [0, 0, 0, 0, 1, 0, 0, 0, 0, 0]),
+    ],
+)
+class TestKillerMoveHeuristic:
+    def test_ordered_moves_end_game(
+        self, fen_string: str, move_scores: list[int]
+    ) -> None:
+        board = init_board(fen_string)
+
+        all_moves = board.legal_moves
+        print(board.legal_moves)
+        num_moves = len(move_scores)
+        assert len(list(all_moves)) == num_moves
+        mo = KillerMoveHeuristic(1)
+        mo.add_killer_move(chess.Move.from_uci("c4b6"), 1)
+
+        for i, move in enumerate(all_moves):
+            score = mo.evaluate(board, move, 1)
+            assert score == move_scores[i]
+
+    def test_sorting_legal_moves(self, fen_string: str, move_scores: list[int]) -> None:
+        board = init_board(fen_string)
+        s: MiniMaxVariants = SearcherFactory.create(
+            SearcherConfig(move_order_mode=MoveOrderMode.KILLER_MOVE)
+        )
+
+        mo: KillerMoveHeuristic = s._move_order
+        mo.add_killer_move(chess.Move.from_uci("c4b6"), 1)
+        legal_moves = s._ordered_moves(board, board.legal_moves, 1)
+
+        for i, move in enumerate(legal_moves):
+            score = mo.evaluate(board, move, 1)
+            assert score == (1 if i == 0 else 0)
```

Comment:
[Minor] This test only works for a unique board - I think using @pytest.mark.parametrize might not be good idea, as it suggest that we can add more inputs for testing

Not blocker

Reply:
fair play, sounds g